lib/galaxy/webapps/galaxy/services/sharable.py

```diff
@@ -117,7 +117,11 @@ def share_with_users(self, trans, id: DecodedDatabaseIdField, payload: ShareWith
         base_status = self._get_sharing_status(trans, item)
         status = self.share_with_status_cls.model_construct(**base_status.model_dump(), extra=extra)
         status.errors.extend(errors)
-        galaxy_url = str(trans.url_builder("/", qualified=True)).rstrip("/") if trans.url_builder else None
+        galaxy_url = (
+            str(trans.url_builder("/", qualified=True)).rstrip("/")
+            if trans.url_builder
+            else trans.app.config.galaxy_external_url
```

Comment:
I think if the admin has explicitly set this - we should default to this over the request based URL.

Reply:
What if the request includes a subdomain ? That was my thinking with using `galaxy_external_url ` as a fallback

Reply:
I hate overriding an explicit admin decision but you're not wrong given how those subdomain Galaxies work. Oh well.